Assets/Scripts/ECS/Events/DeathEventSystem.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using UnityEngine;
+using Unity.Jobs;
+using Unity.Entities;
+using Ecosystem.ECS.Hybrid;
+using Unity.Mathematics;
+using Unity.Transforms;
+using Unity.Collections;
+
+namespace Ecosystem.ECS.Events
+{
+    /// <summary>
+    /// Kills desired entities
+    /// </summary>
+    public class DeathEventSystem : ComponentSystem
+    {
+        protected override void OnUpdate()
+        {
+
+            Entities.WithAll<DeathCommand>().ForEach((Entity entity, /*int entityInQueryIndex,*/
```

Comment:
WithAll<DeathCommand> is unnecessary when you already have it as a parameter in the lambda.
```suggestion
            Entities.ForEach((Entity entity,
```